src/src/PluginStructs/P016_data_struct.cpp

```diff
@@ -26,23 +30,38 @@ void P016_data_struct::convertCommandLines(struct EventStruct *event) {
     log.reserve(80);
 
     log  = F("P016: struct size: ");
-    log += sizeof(CommandLines);
-    log += '/';
     log += sizeof(tCommandLinesV2);
     log += '/';
     log += sizeof(tCommandLines);
     log += F(" enum: ");
     log += sizeof(decode_type_t);
+    log += F(" p016: ");
+    log += sizeof(P016_data_struct);
     addLog(LOG_LEVEL_INFO, log);
     #  endif // ifdef PLUGIN_016_DEBUG
   }
 
   // read V1 data && convert
-  LoadCustomTaskSettings(event->TaskIndex, (uint8_t *)&(CommandLinesV1), sizeof(CommandLinesV1));
+  CommandLinesV1.clear(); // Start fresh
+  int loadOffset = 0;
+
+  for (uint8_t i = 0; i < P16_Nlines; i++) {
+    CommandLinesV1.push_back(tCommandLines());
+    LoadFromFile(SettingsType::Enum::CustomTaskSettings_Type,
+                 event->TaskIndex,
+                 (uint8_t *)&(CommandLinesV1[i]),
+                 sizeof(tCommandLines),
+                 loadOffset);
+    loadOffset += sizeof(tCommandLines);
+  }
+
+  CommandLines.clear(); // Start fresh
   uint16_t codeFlags            = 0;
   uint16_t alternativeCodeFlags = 0;
 
   for (int i = 0; i < P16_Nlines; ++i) {
+    CommandLines.push_back(tCommandLinesV2());
+
     if (safe_strncpy(CommandLines[i].Command, CommandLinesV1[i].Command, P16_Nchars)) { // This should never fail.
```

Comment:
Why create an object for copying? Why not add a parameter to the load function to load the "old" version directly into the new object?
Then the new formatted object can also load (and transition) the old format.

Reply:
Reworked into a conversion constructor